### PR TITLE
Isaac label test

### PR DESCRIPTION
**Pull request by IkeKobby**
- Fixed some failing tests after the release of processors. 
- Crop property rules and tests have been added as well.

### CODE REVIEW

**Comment by IkeKobby** [comment]
@maxaalexeeva , I have created a pr.

File: src/test/scala/org/clulab/habitus/variables/TestVariableReaderLabelBased.scala
Comment (kwalcock) [inline src/test/scala/org/clulab/habitus/variables/TestVariableReaderLabelBased.scala:345]: We are only compiling for Scala 2.12, but if we were also doing 2.11, then this dangling comma would cause a problem.  No need to change, but FYI.

**Review by kwalcock — APPROVED** [review]
It looks very good to me.

File: src/test/scala/org/clulab/habitus/variables/TestVariableReaderLabelBased.scala
Comment (IkeKobby) [reply src/test/scala/org/clulab/habitus/variables/TestVariableReaderLabelBased.scala:345]: Okay, well noted. 

Thank you, @kwalcock

**Comment by kwalcock** [comment]
For some reason the tests had not completed in 5 hours.  I restarted them.

**Comment by IkeKobby** [comment]
> For some reason the tests had not completed in 5 hours. I restarted them.

Okay.

**Comment by kwalcock** [comment]
It looks like github is having some kind of problem.  Tests of other PRs that had previously completed successfully are not running, either.  Everything passes on my computer, but if there is not a rush to merge this, I will wait and see if github fixes itself first.

**Comment by IkeKobby** [comment]
> It looks like github is having some kind of problem. Tests of other PRs that had previously completed successfully are not running, either. Everything passes on my computer, but if there is not a rush to merge this, I will wait and see if github fixes itself first.

@kwalcock Okay sure.

**Comment by maxaalexeeva** [comment]
Going though it now. There will be a few of minor things to adjust before merging.

**Comment by IkeKobby** [comment]
@maxaalexeeva Okay.

File: src/test/scala/org/clulab/habitus/variables/TestEntities.scala
Comment (maxaalexeeva) [inline src/test/scala/org/clulab/habitus/variables/TestEntities.scala:466]: to be honest, I am not sure what the ideal quantity extraction should be: 19.3 kg N and P ha-1 and 21.5 kg N and P ha-1? 19.3 kg ha-1 and 21.5 kg ha-1? This test will probably remain forever failing :D 
I think it is fine to leave it as you have it now :)

File: src/test/scala/org/clulab/habitus/variables/TestEntitiesNorms.scala
Comment (maxaalexeeva) [inline src/test/scala/org/clulab/habitus/variables/TestEntitiesNorms.scala:163]: Please add this note here (that will be for me to fix):
```// fixme: will need to adjust vague season rules in processors to allow for the definite article ```

File: src/test/scala/org/clulab/habitus/variables/TestVariableReaderLabelBased.scala
Comment (maxaalexeeva) [reply src/test/scala/org/clulab/habitus/variables/TestVariableReaderLabelBased.scala:256]: Looks like the test in this test set is not checking the text of the mention, only the norm, so it does not make a difference, but let's still adjust the text to what it's supposed to be.

File: src/test/scala/org/clulab/habitus/variables/TestEntitiesNorms.scala
Comment (maxaalexeeva) [inline src/test/scala/org/clulab/habitus/variables/TestEntitiesNorms.scala:270]: Please adjust the test to have ha-1 in desired (4832 kg ha-1)
In the sentence itself, please adjust the text to not have a space in ha- 1. 
@kwalcock looks like both tika and science parse have issues with processing ha-1 while converting. I will follow up on that in pdf2txt, but I think that should be handled there instead of adjusting rules in processors to account for this.

File: src/test/scala/org/clulab/habitus/variables/TestEntitiesNorms.scala
Comment (maxaalexeeva) [inline src/test/scala/org/clulab/habitus/variables/TestEntitiesNorms.scala:296]: Let's discuss in the lab meeting next week! *this should not stop the merge into main, fine to keep this failing for now

File: src/test/scala/org/clulab/habitus/variables/TestEntitiesNorms.scala
Comment (maxaalexeeva) [reply src/test/scala/org/clulab/habitus/variables/TestEntitiesNorms.scala:213]: Same comment in the TestEntities test for this sentence. 
P.S. I think sometime next week I will try to combine some of the entity tests and entity norm tests, but you do not need to worry about that.

**Review by maxaalexeeva — CHANGES_REQUESTED** [review]
Done! There are a few small things to adjust, but it generally looks good! Thanks for addressing all my requests, @IkeKobby!

File: src/test/scala/org/clulab/habitus/variables/TestEntitiesNorms.scala
Comment (kwalcock) [reply src/test/scala/org/clulab/habitus/variables/TestEntitiesNorms.scala:270]: It would be nice to have the PDF for this.  It doesn't seem like they mess up on all instances of what I suspect is ha<sup>-1</sup>.

File: src/test/scala/org/clulab/habitus/variables/TestEntities.scala
Comment (kwalcock) [reply src/test/scala/org/clulab/habitus/variables/TestEntities.scala:466]: It's interesting to see what you have to deal with.  It looks like they are applying nitrogen at the rate of 19.3 kg/ha and phosphorus at 21.5 kg/ha.

File: src/test/scala/org/clulab/habitus/variables/TestEntitiesNorms.scala
Comment (maxaalexeeva) [reply src/test/scala/org/clulab/habitus/variables/TestEntitiesNorms.scala:270]: You are right. Will send relevant files via email shortly.

File: src/test/scala/org/clulab/habitus/variables/TestEntities.scala
Comment (maxaalexeeva) [reply src/test/scala/org/clulab/habitus/variables/TestEntities.scala:466]: That was my first interpretation but I was led astray by that 'and' between N and P (instead of something like 19.3 kg N and 21.5 kg P). With your comment, I am now sure that is the right interpretation.

File: src/test/scala/org/clulab/habitus/variables/TestEntitiesNorms.scala
Comment (kwalcock) [reply src/test/scala/org/clulab/habitus/variables/TestEntitiesNorms.scala:270]: I put the file through the Adobe converter and also saved from the PDF to text.  Neither method got the hyphens correct.  I converted from PDF to PS and it doesn't look like these hyphens are glyphs at all.  They are lines drawn or something similar.  I'm not sure that pdf2txt is the right place to fix it, because it seems fairly specific to habitus.  There may need to be a regular expression that looks for " ha ?-?1" that converts to " ha-1".

Comment (maxaalexeeva) [reply src/test/scala/org/clulab/habitus/variables/TestEntitiesNorms.scala:270]: Thanks for checking! Yes, that makes sense. I'll adjust it in habitus.

Comment (kwalcock) [reply src/test/scala/org/clulab/habitus/variables/TestEntitiesNorms.scala:270]: On the other hand, the provenance gets messed up if the reader, habitus here, changes the text internally so that positions don't match the text file.  Perhaps I should do something so that the file is corrected and not just the ephemeral string at runtime.

Comment (maxaalexeeva) [reply src/test/scala/org/clulab/habitus/variables/TestEntitiesNorms.scala]: That would be good too! I wonder if this is a one-off thing or there is actually value in handling this weird case in pdf2txt.

**Comment by maxaalexeeva** [comment]
Hi @MihaiSurdeanu, any objections to this being merged?

**Comment by MihaiSurdeanu** [comment]
LGTM @maxaalexeeva ! Thank you!

**Pull request merged**
